### PR TITLE
package: increase minimum vscode to 1.65

### PR DESCRIPTION
**Pull request by justinmk3**
## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/utilities/tsUtils.ts [hunk 1]
```diff
@@ -43,7 +50,7 @@ export function isNonNullable<T>(obj: T): obj is NonNullable<T> {
 }
 
 export function isKeyOf<T>(key: PropertyKey, obj: T): key is keyof T {
-    return key in obj
+    return key in (obj as any)
```
Comment (justinmk3) [inline src/shared/utilities/tsUtils.ts:53]: 🤷

File: src/test/lambda/config/samParameterCompletionItemProvider.test.ts [hunk 2]
```diff
@@ -132,7 +132,7 @@ describe('SamParameterCompletionItemProvider', async function () {
     it('does not provide suggestions if document symbols could not be loaded', async function () {
         const provider = new SamParameterCompletionItemProvider(
             new MockSamParamComplItemProviderContext({
-                executeCommand: async () => undefined,
+                executeCommand: async () => new Promise(() => undefined),
```
Comment (justinmk3) [inline src/test/lambda/config/samParameterCompletionItemProvider.test.ts:135]: 🤷

File: src/test/mockOutputChannel.ts [hunk 3]
```diff
@@ -18,6 +18,11 @@ export class MockOutputChannel implements vscode.OutputChannel {
         return this.onDidAppendTextEmitter.event
     }
 
+    public replace(value: string): void {
+        this._value += value
```
Comment (justinmk3) [inline src/test/mockOutputChannel.ts:22]:
```suggestion
        this._value = value
```

File: src/test/shared/utilities/symbolUtilities.test.ts [hunk 4]
```diff
@@ -41,11 +41,11 @@ describe('symbolUtilities', async function () {
         it('does not retry if maxRetries is 0', async function () {
             const executeCommandArgs: { command: string; uri: vscode.Uri }[] = []
             const context: LoadSymbolsContext = {
-                async executeCommand<T>(command: string, ...args: any[]): Promise<T | undefined> {
+                async executeCommand<T>(command: string, ...args: any[]): Promise<T> {
                     assert.strictEqual(args.length, 1)
                     executeCommandArgs.push({ command, uri: args[0] as vscode.Uri })
 
-                    return undefined
+                    return undefined as T
```
Comment (justinmk3) [inline src/test/shared/utilities/symbolUtilities.test.ts:48]: 🤷

File: src/test/shared/vscode/window.ts [hunk 5]
```diff
@@ -214,7 +214,7 @@ export function createTestWindow(workspace = vscode.workspace): Window & TestWin
         if (Array.isArray(items)) {
             setItems(items)
         } else {
-            items.then(setItems)
+            ;(items as any).then(setItems)
```
Comment (justinmk3) [inline src/test/shared/vscode/window.ts:217]: 👀

File: src/shared/utilities/tsUtils.ts (hunk 1 above)
Comment (JadenSimon) [reply src/shared/utilities/tsUtils.ts:53]: Probably have to narrow `T` with `T extends Record<PropertyKey, any>`. Not sure why we have linting on for `object` as that'd be the most accurate here.

File: src/test/lambda/config/samParameterCompletionItemProvider.test.ts (hunk 2 above)
Comment (JadenSimon) [reply src/test/lambda/config/samParameterCompletionItemProvider.test.ts:135]: Maybe `Promise.resolve(undefined)` ?

File: src/test/shared/vscode/window.ts (hunk 5 above)
Comment (JadenSimon) [reply src/test/shared/vscode/window.ts:217]:
```ts
} else if (items instanceof Promise) {
    items.then(setItems)
} else {
    // Not sure what this type is
    throw new Error('Not implemented ')
}
```

**Review by JadenSimon — APPROVED** [review]
Probably need a changelog for this

File: src/test/lambda/config/samParameterCompletionItemProvider.test.ts (hunk 2 above)
Comment (justinmk3) [reply src/test/lambda/config/samParameterCompletionItemProvider.test.ts:135]: > Type '() => Promise<undefined>' is not assignable to type '<T = unknown>(command: string, ...rest: any[]) => Thenable<T>'

Comment (JadenSimon) [reply src/test/lambda/config/samParameterCompletionItemProvider.test.ts]: 🤷

File: src/codecatalyst/auth.ts [hunk 6]
```diff
@@ -174,41 +174,6 @@ export class CodeCatalystAuthenticationProvider {
...
-        return (this.instance ??= new this(new CodeCatalystAuthStorage(secrets), ctx.globalState))
-    }
-}
-
-/**
- * `secrets` API polyfill for C9.
- *
- * For development only. Do NOT use this for anything else.
- */
-class SecretMemento implements vscode.SecretStorage {
```
Comment (justinmk3) [inline src/codecatalyst/auth.ts:188]: @JadenSimon this is not needed anymore right?

**Comment by justinmk3** [comment]
`E2E Tests - minimum — Build Status - FAILED` logs:

```
  1) Test how this codebase uses the CodeCatalyst API
       Dev Environment functionality
         prompts to install the ssh extension if not available:
     Error: Timed out waiting to install remote SSH extension
      at /codebuild/output/src381537540/src/github.com/aws/aws-toolkit-vscode/src/testE2E/codecatalyst/client.test.ts:210:27
```

Also saw this on https://github.com/aws/aws-toolkit-vscode/pull/3357 , so not related to this PR.

**Pull request merged**